### PR TITLE
DM-15180: Add coordinate covariance to coordinate calculation

### PR DESCRIPTION
**Pull request by cmsaunders**

### CODE REVIEW

**Review by isullivan — APPROVED** [review]
Please store the new coordinate error columns as `lsst.geom.Angle` objects, to match the existing coordinate columns.

File: src/table/wcsUtils.cc [hunk 1]
```diff
@@ -91,33 +91,81 @@ void updateRefCentroids(geom::SkyWcs const &wcs, ReferenceCollection &refList) {
...
         return;
     }
     auto const schema = getSchema(sourceList[0]);
     Point2DKey const centroidKey(schema["slot_Centroid"]);
+    CovarianceMatrixKey<float, 2> const centroidErrKey(schema["slot_Centroid"], {"x", "y"});
     CoordKey const coordKey(schema["coord"]);
     std::vector<lsst::geom::Point2D> pixelList;
     pixelList.reserve(sourceList.size());
+    std::vector<Eigen::Matrix2f> skyErrList;
+    skyErrList.reserve(sourceList.size());
```
[the remark this answers is not in the document]
Comment (TallJimbo) [reply src/table/wcsUtils.cc:135]: I think we'd need to add a new class with `angle^2` units to do that.

[the remark this answers is not in the document]
Comment (isullivan) [reply src/table/wcsUtils.cc:135]: Sorry, my comment was in the wrong place. I don't think we need to store the covariance matrix with units, but when the matrix is used to calculate `coord_raErr` and `coord_decErr`, those columns in the source catalog should have `lsst.geom.Angle` units like `coord_ra` and `coord_dec`.

[the remark this answers is not in the document]
Comment (cmsaunders) [reply src/table/wcsUtils.cc:135]: I think @TallJimbo's comment is still relevant here. The units are set at https://github.com/lsst/afw/blob/0af9017808e0f6d03578c08115bfc95db4e37828/src/table/aggregates.cc#L129, but I don't think the resulting table could or should have `lsst.geom.Angle` units for `coord_raErr` and `coord_decErr` if we don't have the needed units class for `coord_ra_dec_Cov`.

[the remark this answers is not in the document]
Comment (isullivan) [reply src/table/wcsUtils.cc:135]: OK, thank you for explaining why `coord_raErr` and `coord_decErr` can't have units attached.

**Pull request merged**